script.twittertweets/gui.py

```diff
@@ -0,0 +1,94 @@
+#   Copyright (C) 2017 Lunatixz
+#
+#
+# This file is part of Twitter Tweets.
+#
+# Twitter Tweets is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Twitter Tweets is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Twitter Tweets.  If not, see <http://www.gnu.org/licenses/>.
+
+import threading, json
+import xbmc, xbmcaddon, xbmcvfs, xbmcgui
+
+# Plugin Info
+ADDON_ID       = 'script.twittertweets'
+REAL_SETTINGS  = xbmcaddon.Addon(id=ADDON_ID)
+ADDON_NAME     = REAL_SETTINGS.getAddonInfo('name')
+ADDON_VERSION  = REAL_SETTINGS.getAddonInfo('version')
+ADDON_PATH     = (REAL_SETTINGS.getAddonInfo('path').decode('utf-8'))
+SETTINGS_LOC   = REAL_SETTINGS.getAddonInfo('profile').decode('utf-8')
+
+## GLOBALS ##
+CLOSE_TIME = [2.0,5.0,10.0,15.0][int(REAL_SETTINGS.getSetting('Close_Time'))]
+              
+def log(msg, level=xbmc.LOGDEBUG):
+    xbmc.log(ADDON_ID + '-' + ADDON_VERSION + '-' + msg, level)
+      
+def stringify(string):
+    if isinstance(string, list):
+        string = (string[0])
+    elif isinstance(string, (int, float, long, complex, bool)):
+        string = str(string) 
+    
+    if isinstance(string, basestring):
+        if not isinstance(string, unicode):
+            string = unicode(string, 'utf-8')
+        elif isinstance(string, unicode):
+            string = string.encode('ascii', 'ignore')
```

Comment:
As I said, with `.encode('ascii', 'ignore')` all non-ASCII chars will be removed. If you need a proper format for `xbmc.log()`, use `.encode('utf-8')` like in the string below that will never be reached because of a logic error in your if/elif/else construction.
Also, `xbmc.log()` does not accept non-encoded `unicode` objects, so the string above also does not make sense.
The safes option is to store all text as `unicode` and encode it with `utf-8` before passing to Kodi Python API.